saga/sagalogs/file_saga_log.go

```diff
@@ -0,0 +1,387 @@
+	case saga.StartTask.String():
+		taskId, data, err := parseTask(sagaId, scanner)
+		if err != nil {
+			return saga.SagaMessage{}, err
+		}
+		return saga.MakeStartTaskMessage(sagaId, taskId, data), nil
+
+		// Parse End Task Message
+	case saga.EndTask.String():
+		taskId, data, err := parseTask(sagaId, scanner)
+		if err != nil {
+			return saga.SagaMessage{}, err
+		}
+		return saga.MakeEndTaskMessage(sagaId, taskId, data), nil
+
+		// Parse Start Comp Task Message
+	case saga.StartCompTask.String():
+		taskId, data, err := parseTask(sagaId, scanner)
+		if err != nil {
+			return saga.SagaMessage{}, err
+		}
+		return saga.MakeStartCompTaskMessage(sagaId, taskId, data), nil
+
+		// Parse End Comp Task Message
+	case saga.EndCompTask.String():
+		taskId, data, err := parseTask(sagaId, scanner)
+		if err != nil {
+			return saga.SagaMessage{}, err
+		}
+		return saga.MakeEndCompTaskMessage(sagaId, taskId, data), nil
+
+		// Unrecognized Message
+	default:
+		return saga.SagaMessage{}, saga.NewCorruptedSagaLogError(
+			sagaId,
+			fmt.Sprintf("Error Parsing SagaLog unrecognized message type, %v", scanner.Text()),
+		)
+	}
+}
+
+// Helper function that parses a task message, StartTask, EndTask, StartCompTask,
+// EndCompTasks.  Message is of structure
+// line1: MessageType
+// line2: TaskId
+// line3: TaskData FileName
+// Returns a tuple of TaskId, TaskData, Error
+func parseTask(sagaId string, scanner *bufio.Scanner) (string, []byte, error) {
+	// read taskId and datafileName
+	if ok := scanner.Scan(); !ok {
+		return "", nil,
+			saga.NewCorruptedSagaLogError(
+				sagaId,
+				fmt.Sprintf("Error Parsing SagaLog expected TaskId, Error: %v",
+					createUnexpectedScanEndMsg(scanner)),
+			)
+	}
+	taskId := scanner.Text()
+	if ok := scanner.Scan(); !ok {
+		return "", nil,
+			saga.NewCorruptedSagaLogError(
+				sagaId,
+				fmt.Sprintf("Error Parsing SagaLog expected Data, Error: %v",
+					createUnexpectedScanEndMsg(scanner)),
+			)
+	}
+	dataFileName := scanner.Text()
+
+	data, err := ioutil.ReadFile(dataFileName)
+	if err != nil {
+		return "", nil,
+			saga.NewCorruptedSagaLogError(
+				sagaId,
+				fmt.Sprintf("Error Reading DataFile %v, Error: %v", dataFileName, err),
+			)
+	}
+
+	return taskId, data, nil
+}
+
+func createUnexpectedScanEndMsg(scanner *bufio.Scanner) string {
+	var errMsg string
+	if scanner.Err() != nil {
+		errMsg = scanner.Err().Error()
+	} else {
+		errMsg = "Unexpected EOF"
+	}
+	return errMsg
+}
+
+//
+// Returns a list of all in progress sagaIds.
+// This MUST include all not completed sagaIds.
+// It may also included completed sagas
+// Returns an error if it fails.
+//
+// This is a very dumb implementation currently just returns
+// a list of all sagas.  This can be improved with an index
+//
+func (log *fileSagaLog) GetActiveSagas() ([]string, error) {
+	files, err := ioutil.ReadDir(log.dirName)
+	if err != nil {
+		return nil, err
+	}
+
+	sagaIds := make([]string, len(files), len(files))
```

Comment:
You can use the two-arg form of make (the last arg is capacity, which is unneeded if you want size == capacity)